src/llmq/signing_shares.cpp

```diff
@@ -1121,13 +1121,18 @@ bool CSigSharesManager::SendMessages(CConnman& connman)
         auto& nodeState = nodeStates[nodeId];
         auto* session = nodeState.GetSessionBySignHash(signHash);
         assert(session);
-        if (session->sendSessionId == UNINITIALIZED_SESSION_ID) {
-            session->sendSessionId = nodeState.nextSendSessionId++;
-
-            sigSessionAnnouncements[nodeId].emplace_back(
+        while (session->sendSessionId == UNINITIALIZED_SESSION_ID) {
+            const uint32_t session_id{GetRand<uint32_t>()};
```

Comment:
if session_id is randomly generated equal to `uint32_t UNINITIALIZED_SESSION_ID{std::numeric_limits<uint32_t>::max()}` it will cause sigSessionAnnouncements[nodeId] to have added 2 copies for CSigSesAnn with different IDs (one is UNINITIALIZED_SESSION_ID).


```suggestion
            const uint32_t session_id{GetRand<uint32_t>(UNINITIALIZED_SESSION_ID-1)};
```

Reply:
indeed, it's not obvious

Reply:
Yeah... We could move `emplace_back` out of `while` loop maybe. This won't change anything but at least we wouldn't have to worry about duplicates at all. What do you think?